packages/gatsby-source-wordpress/src/steps/source-nodes/update-nodes/content-update-interval.js

```diff
@@ -108,28 +111,51 @@ const refetcher = async (
   )
 }
 
+let startedPolling = false
+let firstCompilationDone = false
+
 /**
  * Starts constantly refetching the latest WordPress changes
  * so we can update Gatsby nodes when data changes
  */
 const startPollingForContentUpdates = helpers => {
   if (
+    startedPolling ||
     process.env.WP_DISABLE_POLLING ||
     process.env.ENABLE_GATSBY_REFRESH_ENDPOINT
   ) {
     return
   }
 
+  startedPolling = true
+
   const { verbose, develop } = store.getState().gatsbyApi.pluginOptions
 
   const msRefetchInterval = develop.nodeUpdateInterval
 
-  if (verbose) {
-    helpers.reporter.log(``)
-    helpers.reporter.info(formatLogMessage`Watching for WordPress changes`)
-  }
-
-  refetcher(msRefetchInterval, helpers)
+  helpers.emitter.on(`COMPILATION_DONE`, () => {
+    /**
+     * we only want to start our refetcher helper 1 time after the first COMPILATION_DONE event.
+     * This event happens when the dev server is ready. It also happens after saving a code change. We only want to run our code 1 time.
+     * onCreateDevServer (the node API we're hooking into) is called before the dev server is ready.
+     * Running our logic at that point is problematic because we could end up triggering the WEBHOOK_RECEIVED event before the dev server is ready and this can cause Gatsby to throw errors. So we're hooking into COMPILATION_DONE to avoid that problem.
+     */
+    if (!firstCompilationDone) {
+      firstCompilationDone = true
+
+      // wait a second so that terminal output is more smooth
+      setTimeout(() => {
+        if (verbose) {
+          helpers.reporter.log(``)
```

Comment:
Do we need this?
```suggestion
```

Reply:
It's not necessary but improves the terminal output readability a bit. When this is logged out the rest of the logs (the part telling you to open localhost:8000) is more spaced out, so this keeps the formatting even.